### PR TITLE
Misc. trivial typos

### PR DESCRIPTION
**Pull request by luzpaz**
Found via `codespell -q 3`

### CODE REVIEW

**Comment by jsquyres** [comment]
Thank you @luzpaz!

Can you please sign off your commit (per https://github.com/open-mpi/ompi/blob/master/.github/CONTRIBUTING.md#contributors-declaration)?  Thanks!

**Comment by luzpaz** [comment]
@jsquyres I'm running `git commit -s` and it's not leaving a signature...? any ideas ?

**Comment by ggouaillardet** [comment]
@luzpaz you need to `git commit --amend -s` and then `git push --force ...`

File: config/opal_check_os_flavors.m4 [hunk 1]
```diff
@@ -22,7 +22,7 @@ AC_DEFUN([OPAL_CHECK_OS_FLAVOR_SPECIFIC],
     AC_MSG_CHECKING([$1])
     AC_COMPILE_IFELSE([AC_LANG_PROGRAM(
      [[#ifndef $1
-      error: this isnt $1
+      error: this isn't $1
```
Comment (bwbarrett) [inline config/opal_check_os_flavors.m4]: This should be `isnt` or `is not`, but not `isn't`.  The ' will cause all kinds of escaping fun in that file.

**Comment by bwbarrett** [comment]
ok to test

**Comment by jsquyres** [comment]
Yeah, I visibly winced at some of the typos I saw fixed in this PR.  😲

File: config/opal_check_os_flavors.m4 (hunk 1 above)
Comment (luzpaz) [reply config/opal_check_os_flavors.m4]: reverted this fix

**Comment by luzpaz** [comment]
There are about roughly 300+ more typos if you want to see them yourself download `codespell` (https://github.com/lucasdemarchi/codespell/). I can submit them in manageable chunks if you prefer ?

**Comment by jsquyres** [comment]
@luzpaz Your 2nd commit looks good, too.  Can you squash into the 1st commit, and have just one (that is signed off)?  Thanks!

**Comment by luzpaz** [comment]
Squashed and rebased. Ready to go

**Pull request merged**